### PR TITLE
Fix #6395: Check space character after "type" and "data" when parsing GHC pragma

### LINKED ISSUE #6395: `dataXXX` identifiers mis-parsed by {-# COMPILE GHC #-}
On Agda 2.6.2.2, this code type-checks but fails GHC compilation with `Failed to parse GHC pragma '= dataCell'`

```agda
open import Agda.Builtin.Nat
postulate x : Nat

{-# FOREIGN GHC
dataCell :: Integer
dataCell = 123456
#-}
{-# COMPILE GHC x = dataCell #-}

```

The compiler error disappears when I rename `dataCell` to anything not starting with `data` or `type`.

### PR DESCRIPTION
**Pull request by szumixie**
Fixes #6395 with the help of @lawcho

### CODE REVIEW

**Review by andreasabel — APPROVED** [review]
Great, thanks!

File: test/Compiler/simple/Issue6395.agda [hunk 1]
```diff
@@ -0,0 +1,8 @@
+open import Agda.Builtin.Nat
+postulate x : Nat
+
+{-# FOREIGN GHC
+dataCell :: Integer
+dataCell = 123456
+#-}
+{-# COMPILE GHC x = dataCell #-}
```
Comment (andreasabel) [inline test/Compiler/simple/Issue6395.agda:11]: Something makes this test fail.  Could be that you need to add a `main` function.  Why not add a `main` function that prints the `dataCell`?

**Review by andreasabel — CHANGES_REQUESTED** [review]
Tests need fixing...

File: test/Compiler/simple/Issue6395.agda (hunk 1 above)
Comment (szumixie) [reply test/Compiler/simple/Issue6395.agda:11]: Got it.

**Review by andreasabel — APPROVED** [review]

**Pull request merged**